src/host/screenInfo.cpp

```diff
@@ -718,9 +718,8 @@ void SCREEN_INFORMATION::SetViewportSize(const COORD* const pcoordSize)
 
         if (_psiMainBuffer)
         {
-            const auto bufferSize = GetBufferSize().Dimensions();
-
-            _psiMainBuffer->SetViewportSize(&bufferSize);
+            _psiMainBuffer->_fAltWindowChanged = false;
+            _psiMainBuffer->_deferredPtyResize = til::size{ GetBufferSize().Dimensions() };

```

Comment:
I don't think this is the right place to be doing this. You're deferring the viewport size, but that's not the same thing as the buffer size. I don't know if this is good enough to make the Terminal alt buffer work (I haven't looked at that side), but in cohost the results are a bit bonkers. For example, try something like this:

```
printf "\e[?1049h" # Switch to alt buffer
printf "\e[8;24;60t" # Make the window narrower (assuming 80x24 shrinking to 60x24)
printf "\e[?1049l" # Switch back to the main buffer
```

I'd expect the main buffer to keep that 60x24 size, but what you get is a 80 width buffer in a 60 width window. In other cases it'll just do nothing at all.

And ideally this would be using the same mechanism as the existing `_fAltWindowChanged` deferred sizing, otherwise there are bound to be weird results when you mix the too. I don't know how feasible that is though.

I honestly don't know what the right answer is here, but this definitely feels wrong.

Reply:
This is exactly the kind of feedback I was looking for. Lemme take a look ☺️

Reply:
(did not find this already on the backlog)

Reply:
Yeah, this didn't work in conhost before because `SetViewportSize` wasn't deferring the size requests in alt mode. But that's exactly what you're doing now, so this update should really fix that bug, except that it doesn't. So my concern is that we'd essentially have to rewrite this code to fix it, so why not just do it right the first time?

That said, I don't know what the right solution is, so if it's not obvious to you how to deal with it, it's probably not a big deal to leave it as is. It's obviously not a common use-case else someone would assumedly have filed a bug report by now.

Reply:
You're not wrong 😛 'spose I was trying to selfishly get these bits of the alt buffer in now, and worry about that bug later. I'll give it another go though.

Reply:
Okay, I think this should be fixed in the latest push now.